### ISSUE TITLE
Connection to qiita.ucsd.edu timed out.

### ISSUE THREAD

**Issue by kattweigh2** [biocore/redbiom#130]
Hi there,

I installed redbiom using conda: conda install -c conda-forge redbiom
To trial, I ran the example command: redbiom search metadata beer | head

But received the error attached. It appears it's timing out trying to connect to qiita.ucsd.edu
Apologies if this is a user/firewall issue but if you could give any suggestion of the cause of the issue that would be great!

Thanks :)

[stderr.txt](https://github.com/biocore/redbiom/files/14505332/stderr.txt)

**Comment by wasade** [comment]
Thanks, could you try the command again? We have occasional timeouts which we haven't traced fully, and the command is working on my end

**Comment by kattweigh2** [comment]
Thanks for your quick response! Just tried again and received the same error. I've tried it multiple times over the past week and have tried on two different HPCs with the same error message every time :/

**Comment by wasade** [comment]
That's odd, I wonder if the systems are blocking outbound on the port used. Can you send the output from `traceroute qiita.ucsd.edu`?

**Comment by kattweigh2** [comment]
traceroute to qiita.ucsd.edu (169.228.46.28), 30 hops max, 60 byte packets
 1  _gateway (10.199.147.21)  1.115 ms * *
 2  10.199.144.22 (10.199.144.22)  0.096 ms  0.096 ms  0.085 ms
 3  ip-10-215-162-193.ap-southeast-2.compute.internal (10.215.162.193)  48.534 ms  48.529 ms  48.523 ms
 4  ip-10-215-168-105.ap-southeast-2.compute.internal (10.215.168.105)  4.217 ms  4.215 ms  4.204 ms
 5  ip-10-215-168-106.ap-southeast-2.compute.internal (10.215.168.106)  3.359 ms  3.337 ms  3.339 ms
 6  ip-10-215-173-126.ap-southeast-2.compute.internal (10.215.173.126)  4.256 ms  4.219 ms  4.261 ms
 7  10.0.32.5 (10.0.32.5)  10.155 ms  5.781 ms  7.891 ms
 8  192.168.67.113 (192.168.67.113)  3.797 ms  3.801 ms  3.783 ms
 9  * * *
10  * * *
11  * * *
12  * * *
13  * * *
14  * * *
15  * * *
16  * * *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
26  * * *
27  * * *
28  * * *
29  * * *
30  * * *

**Comment by kattweigh2** [comment]
`traceroute to qiita.ucsd.edu (169.228.46.28), 30 hops max, 60 byte packets
 1  _gateway (10.199.147.21)  1.115 ms * *
 2  10.199.144.22 (10.199.144.22)  0.096 ms  0.096 ms  0.085 ms
 3  ip-10-215-162-193.ap-southeast-2.compute.internal (10.215.162.193)  48.534 ms  48.529 ms  48.523 ms
 4  ip-10-215-168-105.ap-southeast-2.compute.internal (10.215.168.105)  4.217 ms  4.215 ms  4.204 ms
 5  ip-10-215-168-106.ap-southeast-2.compute.internal (10.215.168.106)  3.359 ms  3.337 ms  3.339 ms
 6  ip-10-215-173-126.ap-southeast-2.compute.internal (10.215.173.126)  4.256 ms  4.219 ms  4.261 ms
 7  10.0.32.5 (10.0.32.5)  10.155 ms  5.781 ms  7.891 ms
 8  192.168.67.113 (192.168.67.113)  3.797 ms  3.801 ms  3.783 ms
 9  * * *
10  * * *
11  * * *
12  * * *
13  * * *
14  * * *
15  * * *
16  * * *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
26  * * *
27  * * *
28  * * *
29  * * *
30  * * *
`

**Comment by kattweigh2** [comment]
traceroute to qiita.ucsd.edu (169.228.46.28), 30 hops max, 60 byte packets
 1  _gateway (10.199.147.21)  1.115 ms * *
 2  10.199.144.22 (10.199.144.22)  0.096 ms  0.096 ms  0.085 ms
 3  ip-10-215-162-193.ap-southeast-2.compute.internal (10.215.162.193)  48.534 ms  48.529 ms  48.523 ms
 4  ip-10-215-168-105.ap-southeast-2.compute.internal (10.215.168.105)  4.217 ms  4.215 ms  4.204 ms
 5  ip-10-215-168-106.ap-southeast-2.compute.internal (10.215.168.106)  3.359 ms  3.337 ms  3.339 ms
 6  ip-10-215-173-126.ap-southeast-2.compute.internal (10.215.173.126)  4.256 ms  4.219 ms  4.261 ms
 7  10.0.32.5 (10.0.32.5)  10.155 ms  5.781 ms  7.891 ms
 8  192.168.67.113 (192.168.67.113)  3.797 ms  3.801 ms  3.783 ms
 9  * * *
10  * * *
11  * * *
12  * * *
13  * * *
14  * * *
15  * * *
16  * * *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
26  * * *
27  * * *
28  * * *
29  * * *
30  * * *

**Comment by wasade** [comment]
Thanks. It looks like your connections bounce around a lot on internal networks. Are you able to, from that host, connect to Qiita?

**Comment by wasade** [comment]
Can you ping google?

**Comment by kattweigh2** [comment]
Hopefully I've done this correctly...

From the command: ping qiita.ucsd.edu
I received:
PING qiita.ucsd.edu (169.228.46.28) 56(84) bytes of data.
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=1 ttl=41 time=151 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=2 ttl=41 time=148 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=3 ttl=41 time=149 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=4 ttl=41 time=148 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=5 ttl=41 time=148 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=6 ttl=41 time=149 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=7 ttl=41 time=150 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=8 ttl=41 time=148 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=9 ttl=41 time=148 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=10 ttl=41 time=149 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=11 ttl=41 time=148 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=12 ttl=41 time=148 ms
64 bytes from qiita.ucsd.edu (169.228.46.28): icmp_seq=13 ttl=41 time=149 ms
etc....

From ping google.com:
PING google.com (142.250.66.206) 56(84) bytes of data.
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=1 ttl=112 tim                                                                              e=6.88 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=2 ttl=112 tim                                                                              e=5.26 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=3 ttl=112 tim                                                                              e=6.93 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=4 ttl=112 tim                                                                              e=5.73 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=5 ttl=112 tim                                                                              e=5.35 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=6 ttl=112 tim                                                                              e=5.29 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=7 ttl=112 tim                                                                              e=5.28 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=8 ttl=112 tim                                                                              e=5.51 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=9 ttl=112 tim                                                                              e=5.31 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=10 ttl=112 ti                                                                              me=5.28 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=11 ttl=112 ti                                                                              me=5.40 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=12 ttl=112 ti                                                                              me=5.57 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=13 ttl=112 ti                                                                              me=5.64 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=14 ttl=112 ti                                                                              me=5.35 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=15 ttl=112 time=5.51 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=16 ttl=112 time=5.34 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=17 ttl=112 time=5.44 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=18 ttl=112 time=5.50 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=19 ttl=112 time=5.50 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=20 ttl=112 time=5.39 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=21 ttl=112 time=5.36 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=22 ttl=112 time=5.88 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=23 ttl=112 time=5.57 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=24 ttl=112 time=5.61 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=25 ttl=112 time=6.75 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=26 ttl=112 time=7.58 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=27 ttl=112 time=5.40 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=28 ttl=112 time=5.24 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=29 ttl=112 time=6.05 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=30 ttl=112 time=5.70 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=31 ttl=112 time=6.67 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=32 ttl=112 time=6.23 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=33 ttl=112 time=5.40 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=34 ttl=112 time=5.51 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=35 ttl=112 time=5.82 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=36 ttl=112 time=5.35 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=37 ttl=112 time=5.34 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=38 ttl=112 time=5.23 ms
64 bytes from syd09s23-in-f14.1e100.net (142.250.66.206): icmp_seq=39 ttl=112 time=5.95 ms
etc...

**Comment by kattweigh2** [comment]
**Extra info...**

**I then got these stats:**

--- qiita.ucsd.edu ping statistics ---
258 packets transmitted, 258 received, 0% packet loss, time 257309ms
rtt min/avg/max/mdev = 147.635/148.431/154.227/0.846 ms

--- google.com ping statistics ---
713 packets transmitted, 712 received, 0.140252% packet loss, time 713133ms
rtt min/avg/max/mdev = 5.200/5.820/26.180/1.603 ms

**And this from curl rather than ping:**
Command: curl -I qiita.ucsd.edu
HTTP/1.1 301 Moved Permanently
Server: nginx/1.24.0
Date: Sun, 10 Mar 2024 22:20:42 GMT
Content-Type: text/html
Content-Length: 169
Connection: keep-alive
Location: https://qiita.ucsd.edu/

Command: curl -I https://qiita.ucsd.edu/
HTTP/1.1 200 OK
Server: nginx/1.24.0
Date: Sun, 10 Mar 2024 22:21:02 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 0
Connection: keep-alive
Etag: "da39a3ee5e6b4b0d3255bfef95601890afd80709"

**Comment by wasade** [comment]
Thanks! Does `redbiom` work from a system that is not part of the HPC environment? It is possible the system administrators are blocking outbound traffic on port 7329. 

Could you send the output of `curl -v http://qiita.ucsd.edu:7329/get/foo` please?

**Comment by kattweigh2** [comment]
curl -v http://qiita.ucsd.edu:7329/get/foo

*   Trying 169.228.46.28:7329...
* connect to 169.228.46.28 port 7329 failed: Connection timed out
* Failed to connect to qiita.ucsd.edu port 7329 after 130926 ms: Connection timed out
* Closing connection 0
curl: (28) Failed to connect to qiita.ucsd.edu port 7329 after 130926 ms: Connection timed out

**Comment by wasade** [comment]
Could you contact the system administrator and inquire about this? You can cc me too if helpful ***@***.***)On Mar 10, 2024, at 17:31, kattweigh2 ***@***.***> wrote:﻿
curl -v http://qiita.ucsd.edu:7329/get/foo

Trying 169.228.46.28:7329...
connect to 169.228.46.28 port 7329 failed: Connection timed out
Failed to connect to qiita.ucsd.edu port 7329 after 130926 ms: Connection timed out
Closing connection 0
curl: (28) Failed to connect to qiita.ucsd.edu port 7329 after 130926 ms: Connection timed out

**Comment by kattweigh2** [comment]
Thanks very much for your help! Have just sent an email with you CC'd. Cheers :)

**Comment by Bas-Voermans** [comment]
I'm inquiring about this issue because i am running into the exact same thing. Was there a solution found?

**Comment by wasade** [comment]
Hi @Bas-Voermans, in this case, outbound traffic on port 7329 was being blocked by @kattweigh2's network administrators. It is also possible a timeout could just occur by chance. Can you try running `redbiom summarize contexts` to see if it triggers a timeout? If it does timeout, then could you email me at "damcdonald AT ucsd.edu"?

**Comment by colinbrislawn** [comment]
I also ran into this issue. As outlined above, here's how to check if redbiom / qiita.ucsd.edu:7329 / is being blocked:

This is how it looks when port 7329 is blocked:
```ssh
$> curl -v http://qiita.ucsd.edu:7329/get/foo

* Host qiita.ucsd.edu:7329 was resolved.
* IPv6: (none)
* IPv4: 169.228.46.28
*   Trying 169.228.46.28:7329...
^C
(I've used ctrl+c to cancel this command)
```

This is how it looks when it's working!

```ssh
$> curl -v http://qiita.ucsd.edu:7329/get/foo

* Host qiita.ucsd.edu:7329 was resolved.
* IPv6: (none)
* IPv4: 169.228.46.28
*   Trying 169.228.46.28:7329...
* Connected to qiita.ucsd.edu (169.228.46.28) port 7329
> GET /get/foo HTTP/1.1
> Host: qiita.ucsd.edu:7329
> User-Agent: curl/8.8.0
> Accept: */*
> 
* Request completely sent off
< HTTP/1.1 200 OK
< Server: Webdis
< Allow: GET,POST,PUT,OPTIONS
< Access-Control-Allow-Methods: GET,POST,PUT,OPTIONS
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: X-Requested-With, Content-Type, Authorization
< Content-Type: application/json
< ETag: "6272c7ec48fb25a0dd654958b7e0ea0f"
< Connection: Keep-Alive
< Content-Length: 12
< 
* Connection #0 to host qiita.ucsd.edu left intact
```

**Comment by wasade** [comment]
Hi @colinbrislawn, can you try again? The infrastructure which supports this site is undergoing maintenance at the moment, but I expect connectivity to exist now.

**Comment by colinbrislawn** [comment]
It's working great for me now!
(I was on library wifi, and switching to my personal hotspot fixed it, leading me to believe it was a firewall and port issue.)
